### PR TITLE
add jdbc url parser for adding for database information to the trace

### PR DESCRIPTION
**Pull request by wuyupengwoaini**

### CODE REVIEW

**Comment by wuyupengwoaini** [comment]
To fix https://github.com/opentracing-contrib/java-spring-cloud/issues/173

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java [hunk 1]
```diff
@@ -88,7 +92,7 @@ static void onError(Throwable throwable, Span span) {
   }
 
   private static Map<String, Object> errorLogs(Throwable throwable) {
-    Map<String, Object> errorLogs = new HashMap<>(2);
+    Map<String, Object> errorLogs = new HashMap<>(3);
```
Comment (malafeev) [inline src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java]: why it is changed from 2 to 3?

**Comment by malafeev** [comment]
tests are failed

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java (hunk 1 above)
Comment (wuyupengwoaini) [reply src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java]: Avoid expansion of capacity

**Comment by wuyupengwoaini** [comment]
> tests failed

fixed

**Comment by malafeev** [comment]
if we 
- don't have a parser for some database

or 
- parser failed due to jdbc driver can support several forms of url

how stable is instrumentation? 
I'm afraid that we can break user's application or spans will contain less information than now.

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java (hunk 1 above)
Comment (malafeev) [reply src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java]: if we add only 2 items, why capacity is 3?

**Comment by wuyupengwoaini** [comment]
Now,
if we  don't have a parser for some database,URLParser wil return a ConnectionInfo.UNKNOWN_CONNECTION_INFO

and I provided a method to the user to register custom parser.

This instrumentation idead is Borrowed from  skywalking,so I think it is stable  enough

File: src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java (hunk 1 above)
Comment (wuyupengwoaini) [reply src/main/java/io/opentracing/contrib/jdbc/JdbcTracingUtils.java]: Because the default load factor is 0.75

**Comment by wuyupengwoaini** [comment]
The safe way of  not breaking user's application is catching the exception of parsing url and return a ConnectionInfo.UNKNOWN_CONNECTION_INFO 

what about you 
@malafeev

**Comment by malafeev** [comment]
I agree, that's the more safe way.

**Comment by wuyupengwoaini** [comment]
I catch all exception but I am not sure to add log or do nothing. And I remove the restrict of adding same parser to let user customer there own parser

**Comment by wuyupengwoaini** [comment]
@malafeev

**Comment by malafeev** [comment]
thanks @wuyupengwoaini, I think we need to add log message using java.util.logging, at least for debugging

**Pull request merged**